### PR TITLE
Remove second order extrapolation.

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
## Main changes of this PR

Removes second order extrapolation.

## Motivation and additional information

The second order extrapolation is rarely used and makes the implementation often very complicated (see below for details). Related to #1089

### Why removing this feature?

The plan is to move the extrapolation feature inside of the class `time::Waveform` to simplify maintainment. However, since the `time::Waveform` does not store data from past windows it is not possible to do a second order extrapolation in the general case. There are two possibilities: Let `time::Waveform` also store data from past windows (with the usual complications with data initialization known from, for example, multistep methods) or remove second order extrapolation.

## Author's checklist

* [x] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [x] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [x] I added a test to cover the proposed changes in our test suite.
* [x] For breaking changes: I documented the changes in the appropriate [porting guide](https://precice.org/couple-your-code-porting-overview.html).
* [x] I sticked to C++17 features.
* [x] I sticked to CMake version 3.16.3.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [x] Do you understand the code changes?

### CODE REVIEW

File: src/cplscheme/BaseCouplingScheme.cpp [hunk 1]
```diff
@@ -73,8 +73,8 @@ BaseCouplingScheme::BaseCouplingScheme(
     PRECICE_ASSERT(_extrapolationOrder == UNDEFINED_EXTRAPOLATION_ORDER, "Extrapolation is not allowed for explicit coupling");
   } else {
     PRECICE_ASSERT(isImplicitCouplingScheme());
-    PRECICE_CHECK((_extrapolationOrder == 0) || (_extrapolationOrder == 1) || (_extrapolationOrder == 2),
-                  "Extrapolation order has to be  0, 1, or 2.");
+    PRECICE_CHECK((_extrapolationOrder == 0) || (_extrapolationOrder == 1),
+                  "Extrapolation order has to be  0 or 1.");
```
Comment (uekerman) [inline src/cplscheme/BaseCouplingScheme.cpp:77]:
```suggestion
                  "Extrapolation order has to be 0 or 1.");
```

File: src/cplscheme/BaseCouplingScheme.hpp [hunk 2]
```diff
@@ -458,9 +458,7 @@ class BaseCouplingScheme : public CouplingScheme {
    *
    * The standard predictor is of order zero, i.e., simply the converged values
    * of the last time windows are taken as initial guess for the coupling iterations.
-   * Currently, an order 1 predictor (linear extrapolation) and order 2 predictor
-   * (see https://doi.org/10.1016/j.compstruc.2008.11.013, p.796, Algorithm line 1 )
-   * is implement besides that.
+   * Currently, an order 1 predictor (linear extrapolation) is implement besides that.
```
Comment (uekerman) [inline src/cplscheme/BaseCouplingScheme.hpp:461]: Please rewrite this comment. Feels like a simple thing is explained in a complicated way.

**Review by uekerman — APPROVED** [review]
Looks good.
Don't forget a changelog entry and a brief remark on the porting guide.

**Comment by BenjaminRodenberg** [comment]
> Don't forget a changelog entry and a brief remark on the porting guide.

Done. https://github.com/precice/precice.github.io/commit/b3c1cc69bb4324b8f7e1e5b7f6667f51544b2757

File: src/cplscheme/BaseCouplingScheme.hpp (hunk 2 above)
Comment (BenjaminRodenberg) [reply src/cplscheme/BaseCouplingScheme.hpp]: Done [cbd8c32](https://github.com/precice/precice/pull/1503/commits/cbd8c329b3a688218805858e71400e042d2319cb)

File: docs/changelog/1503.md [hunk 3]
```diff
@@ -0,0 +1 @@
+* Remove second order extrapolation in acceleration scheme due to unclear use case at high maintainment cost.
```
Comment (uekerman) [inline docs/changelog/1503.md:1]:
```suggestion
* Removed second order extrapolation in acceleration scheme due to unclear use case at high maintainment cost.
```

**Pull request merged**